### PR TITLE
Add support for Anvil supercomputer

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Add support for the Anvil supercomputer at Purdue University. This is an NSF XSEDE machine.

Note: I decided to use Charm++ v7.0.0 immediately.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by knelli2 — DISMISSED** [review]
I've confirmed tests pass and have done several BBH runs without issue.

**Comment by nilsdeppe** [comment]
rebased on develop to get updated test names

**Review by knelli2 — DISMISSED** [review]

File: docs/Installation/InstallationOnClusters.md [hunk 1]
```diff
@@ -35,6 +35,18 @@ want to load the modules, source the shell file for your system and run
 8. Build the targets you are interested in by running, e.g.
    `make -j4 test-executables`
 
+## Anvil at Purdue University
+
+You should build and run tests on a compute node. You can get a compute node by
+running
+```
+sinteractive -N1 -n 10 -p debug -t 60:00
```
Comment (kidder) [inline docs/Installation/InstallationOnClusters.md]: change 10 to 20 so SolveXcts will compile

**Review by kidder — COMMENTED** [review]
you can squash

**Comment by nilsdeppe** [comment]
rebased and squashed. Thanks for testing this out!

**Review by kidder — APPROVED** [review]

**Pull request merged**